library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -378,30 +378,50 @@
     private final ContentResolver contentResolver;
     private final DownloadsUriProvider downloadsUriProvider;
     private final SystemFacade systemFacade;
+    private final BatchRepository batchRepository;
 
     private Uri baseUri;
 
-    public DownloadManager(Context context, ContentResolver resolver) {
-        this(context, resolver, DownloadsUriProvider.getInstance(), new RealSystemFacade(context), false);
+    public static DownloadManager newInstance(Context context, ContentResolver contentResolver) {
```

Comment:
+1

this is not a well known pattern - I'd never seen this before I worked here.